### PR TITLE
[lambda-adapter] Convert lambda adapter to typescript

### PR DESCRIPTION
**Pull request by Naturalclar**
This PR is part of #164 
Converted lambda-adapter to typescript

### CODE REVIEW

File: modules/lambda-adapter/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "build": "upbin tsc --declaration",
+    "ci-test": "upbin nyc npm test --color always",
+    "clean": "upbin rimraf $(cat ../../.temporary-files)",
+    "lint": "upbin eslint 'src/**/*.ts' --fix",
+    "test": "upbin mocha --require ts-node/register 'test/**/*.test.ts' --color always",
+    "type-check": "upbin tsc --noEmit",
+    "watch": "upbin tsc --declaration --watch"
+  },
+  "dependencies": {
+    "@phenyl/interfaces": "~1.0.0",
```
Comment (nicholaslee119) [inline modules/lambda-adapter/package.json]: I think we should take advantage of Lerna,
how about this?
```
@phenyl/interfaces: "*",
@phenyl/utils: "*",
@phenyl/http-rules: "*",
```

File: modules/lambda-adapter/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import { ServerLogic } from "@phenyl/http-rules";
+
+import {
+  HttpMethod,
+  QueryStringParams,
+  ServerParams
+} from "@phenyl/interfaces";
+
+export type LambdaCallback = (err: any, result: any) => void;
```
Comment (nicholaslee119) [inline modules/lambda-adapter/src/index.ts]: there is `Error` type in `@types/node`, should we use it?
`(err: Error | null | undefined, result: any) => void;`

File: modules/lambda-adapter/package.json (hunk 1 above)
Comment (Naturalclar) [reply modules/lambda-adapter/package.json]: version number of lerna monorepo packages are automatically updated using lerna publish.
This command also updates the version number on package.json, so we can continue using the exact version number.
https://github.com/lerna/lerna/tree/master/commands/publish

File: modules/lambda-adapter/src/index.ts (hunk 2 above)
Comment (Naturalclar) [reply modules/lambda-adapter/src/index.ts]: ooh, I missed that. Thanks!
I've modified the types to `(err?: Error | null, result?: any)`

**Pull request merged**